nav2_costmap_2d/include/nav2_costmap_2d/collision_checker.hpp

```diff
@@ -34,9 +36,8 @@
 
 namespace nav2_costmap_2d
 {
-typedef std::vector<geometry_msgs::msg::Point> Footprint;
 
-class CollisionChecker
+class CollisionChecker : public CollisionBase
```

Comment:
Why does this inherit? Wouldn't it be more reasonable for the this subscription-based collision checker to use the collision checker on the given costmap from the subscription? 

In C++ convention is always to compose rather than inherit when possible.